File: todo-rest/src/main/java/com/todorest/entity/Todo.java
package com.todorest.entity;

import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Todo {
  private int id;
  private String userName;
  private String description;
  private boolean done;
  private LocalDate targetDate;
}

File: todo/src/app/service/data/todo.ts
export interface Todo {
  id: number;
  description: string;
  done: boolean;
  targetDate: Date;
}
export class TodoImpl implements Todo {
  constructor(
    public id: number = 0,
    public description: string = '',
    public done: boolean = false,
    public targetDate: Date = new Date()
  ) {}
}
